lib/components/sprite_batch_component.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:ui';
+
+import '../sprite_batch.dart';
+import 'component.dart';
+
+class SpriteBatchComponent extends Component {
+  SpriteBatch spriteBatch;
+  BlendMode blendMode;
+  Rect cullRect;
+  Paint paint;
+
+  SpriteBatchComponent.fromSpriteBatch(
+    this.spriteBatch, {
+    this.blendMode,
+    this.cullRect,
+    this.paint,
+  });
+
+  @override
+  void render(Canvas canvas) {
+    spriteBatch.render(
+      canvas,
+      blendMode: blendMode,
+      cullRect: cullRect,
+      paint: paint,
+    );
+  }
+
+  @override
+  bool loaded() {
+    return spriteBatch != null && spriteBatch.atlas != null;
```

Comment:
you can do `spriteBatch?.atlas != null`. and then you can inline this function like this
```
bool loaded() => spriteBatch?.atlas != null;
```

Reply:
could `loaded` be a `get` property here?